### ISSUE TITLE
`vagrant up` build: multiple errors upgrading packages, installing prereqs, setting up go

### ISSUE THREAD

**Issue by steven-collins-omega** [hashicorp/terraform#9763]
### Expected Behavior
What should have happened?

A Vagrant box with a working build environment should have been set up.

### Actual Behavior
What actually happened?

`go` and related tools are not installed correctly:

```
vagrant@terraform:/opt/gopath/src/github.com/hashicorp/terraform$ make test
==> Checking that code complies with gofmt requirements...
/opt/gopath/src/github.com/hashicorp/terraform/scripts/gofmtcheck.sh: line 5: gofmt: command not found
==> Checking AWS provider for unchecked errors...
==> NOTE: at this time we only look for uncheck errors in the AWS package
==> Installing errcheck...
/opt/gopath/src/github.com/hashicorp/terraform/scripts/errcheck.sh: line 9: go: command not found
/opt/gopath/src/github.com/hashicorp/terraform/scripts/errcheck.sh: line 16: errcheck: command not found
/bin/sh: 2: go: not found
make: *** [generate] Error 127
```

### Steps to Reproduce

Simply follow the instructions in BUILDING.md.

### Important Factoids

Here is the last part of the output of `vagrant up`:

```
==> default: Upgrading packages ...
==> default: W: Size of file /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_trusty-security_main_source_Sources.gz is not what the server reported 152472 152862
==> default: 
==> default: Current default time zone: 'Etc/UTC'
==> default: Local time is now:      Tue Nov  1 03:50:31 UTC 2016.
==> default: Universal Time is now:  Tue Nov  1 03:50:31 UTC 2016.
==> default: Run 'dpkg-reconfigure tzdata' if you wish to change it.
==> default: invoke-rc.d: policy-rc.d denied execution of stop.
==> default: 
==> default: Configuration file '/etc/sudoers'
==> default:  ==> Modified (by you or by a script) since installation.
==> default:  ==> Package distributor has shipped an updated version.
==> default:    What would you like to do about it ?  Your options are:
==> default:     Y or I  : install the package maintainer's version
==> default:     N or O  : keep your currently-installed version
==> default:       D     : show the differences between the versions
==> default:       Z     : start a shell to examine the situation
==> default:  The default action is to keep your current version.
==> default: *** sudoers (Y/I/N/O/D/Z) [default=N] ? 
==> default: 
==> default: Configuration file '/etc/sudoers'
==> default:  ==> Modified (by you or by a script) since installation.
==> default:  ==> Package distributor has shipped an updated version.
==> default:    What would you like to do about it ?  Your options are:
==> default:     Y or I  : install the package maintainer's version
==> default:     N or O  : keep your currently-installed version
==> default:       D     : show the differences between the versions
==> default:       Z     : start a shell to examine the situation
==> default:  The default action is to keep your current version.
==> default: *** sudoers (Y/I/N/O/D/Z) [default=N] ? dpkg: error processing package sudo (--configure):
==> default:  EOF on stdin at conffile prompt
==> default: Errors were encountered while processing:
==> default:  sudo
==> default: E: Sub-process /usr/bin/dpkg returned an error code (1)
==> default: Installing prerequisites ...
Extracting templates from packages: 71%
Extracting templates from packages: 100%
==> default: 
==> default: Configuration file '/etc/sudoers'
==> default:  ==> Modified (by you or by a script) since installation.
==> default:  ==> Package distributor has shipped an updated version.
==> default:    What would you like to do about it ?  Your options are:
==> default:     Y or I  : install the package maintainer's version
==> default:     N or O  : keep your currently-installed version
==> default:       D     : show the differences between the versions
==> default:       Z     : start a shell to examine the situation
==> default:  The default action is to keep your current version.
==> default: *** sudoers (Y/I/N/O/D/Z) [default=N] ? 
==> default: dpkg: error processing package sudo (--configure):
==> default:  EOF on stdin at conffile prompt
==> default: 
==> default: Creating config file /etc/mercurial/hgrc.d/hgext.rc with new version
==> default: Errors were encountered while processing:
==> default:  sudo
==> default: E: Sub-process /usr/bin/dpkg returned an error code (1)
==> default: Downloading go (1.7.3) ...
==> default: Setting up go (1.7.3) ...
==> default: install: cannot stat '/dev/stdin': Stale file handle
```

### References
`git bisect` indicates that this failure was introduced at PR #8762.

**Comment by kwilczynski** [comment]
@steven-collins-omega hi there! I am sorry that your are having issues.

Let me look into fixing this.

**Comment by aseemraj** [comment]
I'm having the exact same issue, in the same sudoers configuration file. Looks like we need an auto confirm flag for reinstalling the config files if it is modified by some script and this option pops up.

**Comment by kwilczynski** [comment]
@aseemraj hi there! Sorry to about the issues!

Yes, precisely, we need to accept upstream configuration changes to the files which are normally set to be protected by the package maintainers.

**Comment by aseemraj** [comment]
I think in the vagrant file, wherever we have apt-get update and upgrade, we can use one of these flags:

1. -o Dpkg::Options::="--force-confnew"
1. -o Dpkg::Options::="--force-confdef"
1. -o Dpkg::Options::="--force-confold"

[This line](https://github.com/hashicorp/terraform/blob/master/Vagrantfile#L19) can include these additional options and it should be good to go as I figure.

**Comment by kwilczynski** [comment]
@steven-collins-omega @aseemraj hi there! Apologies for the delay.

I have opened a Pull Request that contains fixes which should resolve this issue.

**Comment by stack72** [comment]
Closed via #9783
